kolibri/plugins/hooks.py

```diff
@@ -1,59 +1,230 @@
 """
-Hooks are used for your plugin's ``hooks`` attribute, a dictionary
-with hooks you wish to use. The keys are pre-defined constants in this module or arbitrary strings. For instance::
-
-    from kolibri.plugins.base import KolibriPluginBase
-    from kolibri.plugins.hooks import NAVIGATION_POPULATE
-
-    class MyPlugin(KolibriPluginBase):
-
-        def my_navigation(self, navigation_items):
-            navigation_items.append(
-                {
-                    'menu_name': _("My list"),
-                    'menu_url': reverse_lazy("my_plugin:list_of_things")
-                }
-            )
-
-        def hooks(self):
-            return {
-                NAVIGATION_POPULATE: self.my_navigation
-            }
+Kolibri Hooks API
+-----------------
+
+What are hooks
+~~~~~~~~~~~~~~
+
+Hooks are classes that define *something* that happens. It means that you can
+"hook into a component" in Kolibri and have it do a predefined and
+parameterized *thing*. For instance, Kolibri could ask all its plugins who
+wants to add something to the user settings panel, and its then up to the
+plugins to inherit from that specific hook and feed back the parameters that
+the hook definition expects.
+
+The consequences of a hook callback are located anywhere in Kolibri. Each hook
```

Comment:
Hi, nomenclature question for you:

One thing that's confused me a bit is the use of the name "callback".

I normally think of callbacks some piece of code that runs asynchronously after some other code is done, particularly in an event loop. I usually expect that a callback will _not return anything_. E.g. typically I've seen something like:

```python
def longRunningAsyncMethod(param, callback):
    result = doSomethingBig(param)
    callback(result)
```

However, here are some examples from the code base:

```python
    for callback in hooks.get_callables(hooks.NAVIGATION_POPULATE):
        for item in callback():
            yield item
```

```python
    for callback in hooks.get_callables(hooks.FRONTEND_PLUGINS):
        bundle_path, stats_file, async_events = callback()
```

```python
    tags = [render_as_async(callback()) for callback in hooks.get_callables(hook)]
    return mark_safe('\n'.join(tags))
```

They all return some value and are used synchronously by something else. Are they more "getter methods" than "callbacks", perhaps?